libs/bare/basic.ts

```diff
@@ -1,16 +1,16 @@
 namespace control {
     //% shim=serial::writeString
-    export function __log(s: string) {
+    export function __log(s: any) {
```

Comment:
the C++ shim expects a string. You should convert to string before passing to __log.